launcher/minecraft/launch/LauncherPartLaunch.cpp

```diff
@@ -157,7 +164,178 @@ void LauncherPartLaunch::executeTask()
+                bwrapArgs << "--ro-bind-try" << x11Socket << x11Socket;
+            }
+
+            bwrapArgs << "--ro-bind-try" << xAuthPath << defaultXAuthPath;
+            bwrapArgs << "--unsetenv" << "XAUTHORITY";
+
+            if (wlDisplay.startsWith('/'))
+                bwrapArgs << "--ro-bind-try" << wlDisplay << wlDisplay;
+            else
+                bwrapArgs << "--ro-bind-try" << FS::PathCombine(actualRuntimeDir, wlDisplay) << FS::PathCombine(sandboxedRuntimeDir, wlDisplay);
+        }
+
+        if (minecraftInstance->settings()->get("EnableMangoHud").toBool() && APPLICATION->capabilities() & Application::SupportsMangoHud) {
+            auto mangoHudConfigPath = qEnvironmentVariable("MANGOHUD_CONFIGFILE",
+                    FS::PathCombine(QStandardPaths::writableLocation(QStandardPaths::GenericConfigLocation), "MangoHud")); 
+            bwrapArgs << "--ro-bind-try" << mangoHudConfigPath << mangoHudConfigPath;
+        }
+
+        // launcher
+        {
+            QString instPath = QDir::toNativeSeparators(QDir(minecraftInstance->instanceRoot()).absolutePath());
+            QString assetsPath = QDir::toNativeSeparators(QDir("assets").absolutePath());
+            QString librariesPath = QDir::toNativeSeparators(QDir("libraries").absolutePath());
+
+            bwrapArgs << "--bind" << instPath << instPath;
+            bwrapArgs << "--ro-bind" << assetsPath << assetsPath;
+            bwrapArgs << "--ro-bind" << librariesPath << librariesPath;
+
+            // also add NewLaunch.jar just to be safe. This is probably covered by /usr or LINUX_BWRAP_EXTRA_ARGS though
+            bwrapArgs << "--ro-bind" << jarPath << jarPath;
+        }
+
+        bwrapArgs << "--setenv" << "MINECRAFT_SANDBOX_VARIANT" << QString("%1-bwrap").arg(BuildConfig.LAUNCHER_APP_BINARY_NAME);
+
+        // Make xdg-open work
+        bwrapArgs << "--bind" << dbusProxyPath << sandboxedDbusProxyPath;
+        bwrapArgs << "--setenv" << "DBUS_SESSION_BUS_ADDRESS" << QString("unix:path=%1").arg(sandboxedDbusProxyPath);
+        // HACK: Workaround to force xdg-open to use XDG Portals
+        bwrapArgs << "--unsetenv" << "XDG_CURRENT_DESKTOP";
+        bwrapArgs << "--setenv" << "DE" << "flatpak"; // this should skip checks in xdg-utils-common for other DEs
+
+        // TODO: add args from environment variable?
+        bwrapArgs << Commandline::splitArgs(minecraftInstance->settings()->get("BwrapExtraArgs").toString());
+
+        bwrapArgs << "--";
+        args = bwrapArgs + args;
+
+        // Prepare xdg-dbus-proxy
+        // TODO: also wrap this using bwrap?
+        QStringList dbusProxyArgs{xdgDbusProxyPath};
+        dbusProxyArgs << hostDbusPath;
+        dbusProxyArgs << dbusProxyPath;
+        dbusProxyArgs << "--filter";
+        dbusProxyArgs << "--fd=1"; // so we can know once it's ready
+        dbusProxyArgs << "--call=org.freedesktop.portal.*=*";
```

Comment:
Is this too relaxed? `org.freedesktop.portal.Documents` supports file transfer and `org.freedesktop.portal.Desktop` contains a lot of functionalities.
```
$ dbus-send --session --dest=org.freedesktop.portal.Desktop --type=method_call --print-reply "/org/freedesktop/portal/desktop" org.freedesktop.DBus.Introspectable.Introspect | grep 'interface name\|method name'
```
<details>
<summary>All methods</summary>

```
  <interface name="org.freedesktop.DBus.Properties">
    <method name="Get">
    <method name="GetAll">
    <method name="Set">
  <interface name="org.freedesktop.DBus.Introspectable">
    <method name="Introspect">
  <interface name="org.freedesktop.DBus.Peer">
    <method name="Ping"/>
    <method name="GetMachineId">
  <interface name="org.freedesktop.portal.Inhibit">
    <method name="Inhibit">
    <method name="CreateMonitor">
    <method name="QueryEndResponse">
  <interface name="org.freedesktop.portal.Background">
    <method name="RequestBackground">
    <method name="SetStatus">
  <interface name="org.freedesktop.portal.Location">
    <method name="CreateSession">
    <method name="Start">
  <interface name="org.freedesktop.portal.Notification">
    <method name="AddNotification">
    <method name="RemoveNotification">
  <interface name="org.freedesktop.portal.Screenshot">
    <method name="Screenshot">
    <method name="PickColor">
  <interface name="org.freedesktop.portal.Account">
    <method name="GetUserInformation">
  <interface name="org.freedesktop.portal.NetworkMonitor">
    <method name="GetAvailable">
    <method name="GetMetered">
    <method name="GetConnectivity">
    <method name="GetStatus">
    <method name="CanReach">
  <interface name="org.freedesktop.portal.Print">
    <method name="Print">
    <method name="PreparePrint">
  <interface name="org.freedesktop.portal.Settings">
    <method name="ReadAll">
    <method name="Read">
  <interface name="org.freedesktop.portal.GameMode">
    <method name="QueryStatus">
    <method name="RegisterGame">
    <method name="UnregisterGame">
    <method name="QueryStatusByPid">
    <method name="RegisterGameByPid">
    <method name="UnregisterGameByPid">
    <method name="QueryStatusByPIDFd">
    <method name="RegisterGameByPIDFd">
    <method name="UnregisterGameByPIDFd">
  <interface name="org.freedesktop.portal.RemoteDesktop">
    <method name="CreateSession">
    <method name="SelectDevices">
    <method name="Start">
    <method name="NotifyPointerMotion">
    <method name="NotifyPointerMotionAbsolute">
    <method name="NotifyPointerButton">
    <method name="NotifyPointerAxis">
    <method name="NotifyPointerAxisDiscrete">
    <method name="NotifyKeyboardKeycode">
    <method name="NotifyKeyboardKeysym">
    <method name="NotifyTouchDown">
    <method name="NotifyTouchMotion">
    <method name="NotifyTouchUp">
  <interface name="org.freedesktop.portal.MemoryMonitor">
  <interface name="org.freedesktop.portal.OpenURI">
    <method name="OpenURI">
    <method name="OpenFile">
    <method name="OpenDirectory">
  <interface name="org.freedesktop.portal.Realtime">
    <method name="MakeThreadRealtimeWithPID">
    <method name="MakeThreadHighPriorityWithPID">
  <interface name="org.freedesktop.portal.DynamicLauncher">
    <method name="Install">
    <method name="PrepareInstall">
    <method name="RequestInstallToken">
    <method name="Uninstall">
    <method name="GetDesktopEntry">
    <method name="GetIcon">
    <method name="Launch">
  <interface name="org.freedesktop.portal.Wallpaper">
    <method name="SetWallpaperURI">
    <method name="SetWallpaperFile">
  <interface name="org.freedesktop.portal.Camera">
    <method name="AccessCamera">
    <method name="OpenPipeWireRemote">
  <interface name="org.freedesktop.portal.Device">
    <method name="AccessDevice">
  <interface name="org.freedesktop.portal.PowerProfileMonitor">
  <interface name="org.freedesktop.portal.ScreenCast">
    <method name="CreateSession">
    <method name="SelectSources">
    <method name="Start">
    <method name="OpenPipeWireRemote">
  <interface name="org.freedesktop.portal.Email">
    <method name="ComposeEmail">
  <interface name="org.freedesktop.portal.Trash">
    <method name="TrashFile">
  <interface name="org.freedesktop.portal.ProxyResolver">
    <method name="Lookup">
  <interface name="org.freedesktop.portal.FileChooser">
    <method name="OpenFile">
    <method name="SaveFile">
    <method name="SaveFiles">
```

</details>